sympy/polys/ring_series.py

```diff
@@ -1281,7 +1281,7 @@ def rs_acos(p,x,prec):
         p1 = rs_nth_root(p1, -2, x, prec - 1)
         p1 = rs_mul(dp, p1, x, prec - 1)
         p1 = p1*(-1)
-        return rs_integrate(p1, x)
+        return rs_integrate(p1, x) + mp.pi/2
```

Comment:
`acos(0)` is defined in terms of SymPy `pi`.
https://github.com/sympy/sympy/blob/fb98dbb5a50dc13489b365ea6558e1efc5e62377/sympy/functions/elementary/trigonometric.py#L2110-L2111

Reply:
Alright,thank you. I want to add it to ring_series. I will use SymPy rather than mpmath

Reply:
So,converting a constant term to a ring element or a Polyelement has not been implemented yet. As in, ``rs_sin`` the ``c_expr`` is fed into ``sin`` and ``cos`` . But showing an output with a constant term such as pi/2 is not possible. I have been trying to do the same but no success.

Reply:
It should be possible to add `pi` to the coefficient domain of R by something like this:

Extend `R.domain` by unifying it with the ring `ZZ[pi]` generated by `pi`: `dom = R.domain.unify(ZZ[pi])`.
Extend `R` by changing its domain: `Rnew = R.clone(domain=dom)`.
Change the ring of the polynomial `p = p.set_ring(Rnew)`.

Reply:
I wrote this 
```
def rs_acos(p, x, prec):
    if rs_is_puiseux(p, x):
        return rs_puiseux(rs_acos, p, x, prec)
    if _has_constant_term(p, x):
        raise NotImplementedError("Polynomial must not have constant term in "
                                  "series variables")
    R = p.ring
    pi = symbols('pi')
    dom = R.domain.unify_with_symbol(R[x, pi], symbols)
    R = R.clone(domain = dom)
    p = p.set_ring(R)
    k1 = rs_asin(p, x, prec)
    k = R(pi/2)
    if x in R.gens:
        return k - k1
    else:
        raise NotImplementedError
```

getting an error as 
```
Traceback (most recent call last):
  File "/home/chad7/sympy/sympy/polys/tests/test_ring_series.py", line 284, in test_acos
    assert rs_acos(x**2,x,2) == S.Pi/2 - x**2 - x**6/6
  File "/home/chad7/sympy/sympy/polys/ring_series.py", line 1281, in rs_acos
    dom = R.domain.unify(R[x, pi], symbols)
  File "/home/chad7/sympy/sympy/polys/rings.py", line 447, in __getitem__
    symbols = self.symbols[key]
TypeError: tuple indices must be integers or slices, not tuple
```
@jksuom

Reply:
It should not be necessary to redefine `pi` as a symbol. In fact, its value is then lost.

This is wrong: `dom = R.domain.unify(R[x, pi], symbols)`. No additional symbols are needed. Besides, it looks like `symbols` could be a SymPy function and not a list.

The error message comes from `R[x, pi]` that looks like a matrix element.

Reply:
Well, at first I wrote `` dom = R.domain.unify(QQ[S.Pi]) `` but it showed an error ``x in R.gens is not a tupled index`` which means when ``rs_asin `` is called it takes ``x`` into consideration and checks if x is in ``domain`` . Hence,we need to add ``symbols`` . Moreover, other error came as ``is_EX is not an attribute of S.Pi`` So,keeping the value will create an error as we can't use both the domains simultaneously,we need to show ``pi`` as ``symbol`` rather than its value.
It wiil basically be same. And in addition to that ``QQ[S.Pi],symbols`` function isn't iterable.

I think keeping ``unify_with_symbols`` will be good but the parameters I can't decide. @jksuom